src/core/geometry/qgsmultilinestring.cpp

```diff
@@ -178,3 +179,39 @@ bool QgsMultiLineString::wktOmitChildType() const
   return true;
 }
 
+QgsMultiLineString *QgsMultiLineString::measuredLine( double start, double end ) const
+{
+
+
+  QgsMultiLineString *result{createEmptyWithSameType()};
+
+  if ( !result->convertTo( QgsWkbTypes::addM( mWkbType ) ) )
+  {
+    return result;
+  }
+
+  if ( isEmpty() )
+  {
+    return result;
+  }
```

Comment:
There's no benefit of doing things this way vs just creating a new QgsMultiLineString, as the first call to addGeometry for an empty multilinestring automatically sets the multilinestring z/m presence to match the first added geometry:

```suggestion
  std::unique_ptr< QgsMultiLineString > result = std::make_unique< QgsMultiLineString >();)
  if ( isEmpty() )
  {
    result->convertTo( QgsWkbTypes::addM( mWkbType ) )
    return result;
  }
```